tests/benchmark/test_worst_memory.py

```diff
@@ -77,6 +77,15 @@ def test_worst_calldatacopy(
     if size == 0 and non_zero_data:
         pytest.skip("Non-zero data with size 0 is not applicable.")
 
+    # If `non_zero_data` is True, we fill the calldata with deterministic random data.
+    # Note that if `size == 0` and `non_zero_data` is a skipped case.
+    data = Bytes([i % 256 for i in range(size)]) if non_zero_data else Bytes()
+
+    intrinsic_gas_calculator = fork.transaction_intrinsic_cost_calculator()
+    min_gas = intrinsic_gas_calculator(calldata=data)
+    if min_gas > env.gas_limit:
+        pytest.skip("Minimum gas required for calldata ({min_gas}) is greater than the gas limit")
```

Comment:
Theoretically, there could be other gas limits where this check isn't enough. But practically speaking, the low targets are 1M and 5M, so IMO this is good enough without getting into more complex calculations.